src/java/org/apache/cassandra/locator/ReplicaPlans.java

```diff
@@ -352,7 +352,8 @@ public static ReplicaPlan.ForTokenWrite forWrite(Keyspace keyspace, ConsistencyL
         assert liveAndDown.replicationStrategy() == live.replicationStrategy()
                : "ReplicaLayout liveAndDown and live should be derived from the same replication strategy.";
         AbstractReplicationStrategy replicationStrategy = liveAndDown.replicationStrategy();
-        EndpointsForToken contacts = selector.select(consistencyLevel, liveAndDown, live);
+        IEndpointSnitch endpointSnitch = DatabaseDescriptor.getEndpointSnitch();
+        EndpointsForToken contacts = selector.select(consistencyLevel, liveAndDown, live).filter(endpointSnitch.filterByAffinity(keyspace.getName()));
```

Comment:
This _seems_  (I will investigate) to apply the filtering to all writes - breaking the consistency requirements during bootstrap as non-ready nodes would be filtered out in CNDB - while it's only needed for reads and counter forwarding.

Reply:
So the byteman rule is only intended for reads? I can make that work and we can just ignore this

Reply:
So I've investigated the new C* code and `filterByAffinity()` is correctly invoked for both reads and counters (via `StorageProxy#findSuitableReplica`, which is not great but does the job, I will open a followup), so there's no other change required than the byteman rule. I've also verified the integration test passes without any of these changes.